### PR TITLE
Fixes #10659 - template url fetcher error caught correctly

### PR DESCRIPTION
**Pull request by lzap**
Quick merge again? :-)

### CODE REVIEW

File: lib/proxy_api/template.rb [hunk 1]
```diff
@@ -11,12 +11,12 @@ def template_url
         return response["templateServer"]
       end
     rescue SocketError, Timeout::Error, Errno::EINVAL, Errno::ECONNRESET,
```
Comment (domcleal) [inline lib/proxy_api/template.rb]: This bothers me, do you know why we even need/encourage this behaviour?

Comment (lzap) [inline lib/proxy_api/template.rb]: I don't understand. You don't like the amount of exceptions here? Or what in particular?

This call is used to fetch the `template_url` configuration value from proxy. I was not running it and got ugly error, nicer one with this patch.

Comment (domcleal) [inline lib/proxy_api/template.rb]: I don't think I like catching the exception at all.  It looks like we would silently ignore the problem?  I don't see where an error is generated, only logging.

Comment (lzap) [inline lib/proxy_api/template.rb]: This code was commited as part of the 2094e4e8b049e and this concern should have been discussed there, but okay. I can rethrow the exception with a nicer error message.

The issue here is this is used in various contexts (template rendering, bootdisk rake tasks) and bare error message like "Connection refused!" says nothing. This is what I am trying to improve.

**Comment by lzap** [comment]
Wait, I can't just throw because the code offers a fallback mechanism. Rebased.

File: lib/proxy_api/template.rb (hunk 1 above)
Comment (domcleal) [inline lib/proxy_api/template.rb]: Yes, it should have been discussed better, I think the whole feature's terrible.

I don't though wish to accept patches that worsen or entrench what's already a bad piece of code, so yes, I think re-throwing it with a more appropriate message would be best, unless there's a good reason we need to ignore the error.

The only reason I can think of is to try and avoid problems changing a host's build state when the proxy is unavailable, but in that case you're just going to create problems by writing the wrong URL to the templates... and in any case, shouldn't the proxy be available by definition?  Or is this some workaround for its single-threaded nature?

**Comment by tbrisker** [comment]
@lzap,
Thank you very much for your efforts!
This PR has been open for a very long time with no activity.
Please indicate if you plan to continue working on it or not; if not, feel free to close it.
If you do not respond either way within a couple of weeks, I will assume this PR is abandoned and will go ahead and close it for you.

**Comment by lzap** [comment]
Ok to be honest, I don't understand why @GregSutcliffe listed all the exceptions explicitly. Can't we just catch them all? I change the patch to do so...

**Comment by domcleal** [comment]
There's an unresolved point in the outdated diff section, this will cause templates to be written with incorrect URLs (i.e. Foreman, not the smart proxy with the templates feature).

File: lib/proxy_api/template.rb [hunk 2]
```diff
@@ -10,9 +10,7 @@ def template_url
       if (response = parse(get("templateServer"))) and response["templateServer"].present?
         return response["templateServer"]
       end
-    rescue SocketError, Timeout::Error, Errno::EINVAL, Errno::ECONNRESET,
-           EOFError, Net::HTTPBadResponse, Net::HTTPHeaderSyntaxError,
-           Net::ProtocolError, RestClient::ResourceNotFound => e
+    rescue Exception => e
```
Comment (ares) [inline lib/proxy_api/template.rb:13]: We shouldn't `rescue Exception`, that would catch even things like SyntaxError, SystemExit. We should rescue StandardError or avoid using the exception class which defaults to StandardError. I prefer listing only exceptions that I expect so I think this patch does not fix the problem correctly. I'd recommend adding the exception that that was not rescued to the list instead.

**Pull request closed without merging**